apps/search/js/providers/places.js

```diff
@@ -275,28 +204,23 @@
 
   function formatPlace(placeObj, filter) {
 
-    var bookmarkData = {
-      id: placeObj.url,
-      name: placeObj.title || placeObj.url,
-      url: placeObj.url
+    var result = {
+      'title': placeObj.title,
+      'meta': placeObj.url,
+      'dataset': {
+        'url': placeObj.url
+      },
+      'label': placeObj.title
     };
```

Comment:
Instead of removing the call to `getIcon`, do an 

  'icon': getIcon(placeObj)

Here, then in `provider.js`, you can set the src of the icon when its not null